### PR TITLE
Emailwindow

### PR DESCRIPTION
**Pull request by hm2602**
Creates message box that can send email at bottom of user profile.

### CODE REVIEW

**Comment by dragonkittymeowr** [comment]
@francesar

File: app/views.py [hunk 1]
```diff
@@ -114,41 +113,36 @@ def edit():
...
 
 
-@app.route('/user/<uid>', methods=['GET'])
+@app.route('/user/<uid>', methods=['GET', 'POST'])
 @logged_in # eventually only require login if make_public == false
 def user(uid):
     try:
-        user = db.child('users').child(uid).get(session['idToken']).val()
+        viewed_user = db.child('users').child(uid).get(session['idToken']).val()
+        mail = Mail(app)
```
Comment (alanhdu) [inline app/views.py]: Should this be imported?

File: app/views.py [hunk 2]
```diff
@@ -114,41 +113,36 @@ def edit():
...
         profile = db.child('profiles').child(uid).get(session['idToken']).val()
+        form = ContactForm(request.form)
+        user = db.child('users').child(session['uid']).get(session['idToken']).val()
         if user is None or profile is None:
             return render_template('error/404.html', logged_in=True)
```
Comment (alanhdu) [inline app/views.py]: If the `user is None`, maybe we should return a permission denied error or something instead?

File: app/views.py [hunk 3]
```diff
@@ -114,41 +113,36 @@ def edit():
...
+            if form.validate() == False:
+                flash(u'Please write a message', 'danger')
+                return render_template('user.html', viewed_user=viewed_user, profile=profile, logged_in=True, form = form, user= user)
+            else: 
+                msg = Message(subject='Coffee@CU Email', sender='coffeeatcu@gmail.com', recipients=[viewed_user['email']])
+                msg.body = """
+                New Message from %s %s
+                %s
+                %s
+                """ % (user['firstname'], user['lastname'], user['email'], form.message.data)
```
Comment (alanhdu) [inline app/views.py]: This email body's going to have really weird indentation as-is.

File: app/views.py [hunk 4]
```diff
@@ -114,41 +113,36 @@ def edit():
...
+                mail.send(msg)
```
Comment (alanhdu) [inline app/views.py]: So I feel like I've said this before (and apologies if I'm repeating myself), but:

In general, you don't want operations like "send an email" to be synchronous. Sending email is slow and error-prone (and sometimes you can get "email not found" errors *minutes* after you sent it), so you don't want it blocking the main app logic. Generally, you use some kind of asynchronous task queue instead.

**Pull request merged**